measure.rb

```diff
@@ -2805,21 +2806,33 @@ def self.add_ceiling_fans(runner, model, building, weather)
                             @cfa, @living_space)
   end
 
-  def self.check_distribution_system(building, system_values)
-    dist_id = system_values[:distribution_system_idref]
+  def self.check_distribution_system(building, dist_id, system_id, system_type)
     return if dist_id.nil?
 
+    hvac_distribution_type_map = { "Furnace" => ["AirDistribution", "DSE"],
+                                   "Boiler" => ["HydronicDistribution", "DSE"],
+                                   "central air conditioner" => ["AirDistribution", "DSE"],
+                                   "air-to-air" => ["AirDistribution", "DSE"],
+                                   "mini-split" => ["AirDistribution", "DSE"],
+                                   "ground-to-air" => ["AirDistribution", "DSE"] }
+
     # Get attached distribution system
     found_attached_dist = false
+    type_match = true
     building.elements.each("BuildingDetails/Systems/HVAC/HVACDistribution") do |dist|
       hvac_distribution_values = HPXML.get_hvac_distribution_values(hvac_distribution: dist)
       next if dist_id != hvac_distribution_values[:id]
 
       found_attached_dist = true
+      if not hvac_distribution_type_map[system_type].nil?
+        type_match = hvac_distribution_type_map[system_type].include? hvac_distribution_values[:distribution_system_type]
```

Comment:
Why do we need to check for nil here? What systems aren't we handling? My inclination is that we should throw an error if we don't have a system included in the map that can be ducted, but maybe I'm missing a nuance here.

Reply:
No need to check nil here. Removed. 
Distribution counts are restricted in EPvalidator, so no possibility to have an illegal ducted system specified in HPXML.